File: mobile/lingo_cross_app/lib/features/games/domain/game_type.dart
/// Oyun türü (API `GameType` enum'u ile birebir; sayısal değerler kalıcı).
///
/// MVP'de yalnız [wordMatching] kullanılır; diğerleri Faz 2 için rezerve.
enum GameType {
  wordMatching(1),
  crossword(2),
  questionSet(3);

  const GameType(this.value);

  final int value;

  static GameType fromValue(int value) => switch (value) {
        1 => GameType.wordMatching,
        2 => GameType.crossword,
        3 => GameType.questionSet,
        _ => GameType.wordMatching,
      };
}

/// Oyun oturumu durumu (API `GameSessionStatus` enum'u ile birebir).
enum GameSessionStatus {
  inProgress(1),
  completed(2),
  abandoned(3);

  const GameSessionStatus(this.value);

  final int value;

  static GameSessionStatus fromValue(int value) => switch (value) {
        1 => GameSessionStatus.inProgress,
        2 => GameSessionStatus.completed,
        3 => GameSessionStatus.abandoned,
        _ => GameSessionStatus.inProgress,
      };
}

/// Bulmaca kelime yönü (API `CrosswordDirection` enum'u ile birebir).
///
/// [across] (yatay) → harfler sütun artarak, [down] (dikey) → satır artarak yerleşir.
enum CrosswordDirection {
  across(0),
  down(1);

  const CrosswordDirection(this.value);

  final int value;

  static CrosswordDirection fromValue(int value) => switch (value) {
        0 => CrosswordDirection.across,
        1 => CrosswordDirection.down,
        _ => CrosswordDirection.across,
      };
}

File: api/src/LingoCross.Domain/Enums/GameType.cs
namespace LingoCross.Domain.Enums;

/// <summary>
/// Bir oyunun türü. Sayısal değerler kalıcı olarak saklandığı için DEĞİŞTİRİLMEMELİDİR.
/// MVP'de yalnızca <see cref="WordMatching"/> kullanılır; diğerleri Faz 2 için rezervedir.
/// </summary>
public enum GameType
{
    /// <summary>Kelime eşleştirme (kaynak terim ↔ hedef çeviri). MVP.</summary>
    WordMatching = 1,

    /// <summary>Bulmaca (crossword). Faz 2 — rezerve.</summary>
    Crossword = 2,

    /// <summary>Çıkmış soru seti. Faz 2 — rezerve.</summary>
    QuestionSet = 3,

    /// <summary>Harf karıştırma (scrambled): kaynak terimin harfleri karışık verilir, öğrenci doğru sıraya dizer.</summary>
    Scrambled = 4,
}
